app/models/course.rb

```diff
@@ -5,6 +5,14 @@ class Course < ActiveRecord::Base
                                OR lower(title) like ?",
                                "%#{q.to_s.downcase.gsub(/\s+/,'')}%", "%#{q.to_s.downcase}%") }
 
+  # Get all prereq info from the database.
+  # Return an array of arrays of courses.
+  def prerequisites
+    Prerequisite.where(postrequisite: self).reduce([]) do |arr, prereq|
+      arr << Course.where('id IN (?)', prereq.prerequisite_ids)
+    end
+  end
```

Comment:
I'm not sure if that will work with it being an array in the database...I'll look into it though